src/irmin-pack/conf.mli

```diff
@@ -17,6 +17,13 @@
 module type S = sig
   val entries : int
   val stable_hash : int
+
+  type index :=
+    [ `Seeded_hash  (** use a non-crypto seeded-hash of the step *)
+    | `Hash_bits  (** crypto hash the step and extract the relevant bits. *)
+    | `Custom of depth:int -> bytes -> int  (** use a custom index *) ]
+
+  val index : index
```

Comment:
With Irmin 3.0, this module type will also contain the users' indexing strategy (https://github.com/mirage/irmin/pull/1664), which is currently just `val indexing_strategy`. Can we call this something other than `index`, as we already have a notion of "index" in the public API? (maybe `inode_child_order`?)